includes/class-wc-google-analytics.php

```diff
@@ -536,193 +422,6 @@ private function disable_tracking( $type ) {
-		return $url;
-	}
-
-	/**
-	 * Google Analytics event tracking for loop add to cart
-	 */
-	public function loop_add_to_cart() {
-		if ( $this->disable_tracking( $this->ga_event_tracking_enabled ) || 'yes' === $this->ga_gtag_enabled ) {
-			return;
-		}
-
-		// Add single quotes to allow jQuery to be substituted into _trackEvent parameters
-		$parameters             = array();
-		$parameters['category'] = "'" . __( 'Products', 'woocommerce-google-analytics-integration' ) . "'";
-		$parameters['action']   = "'" . __( 'Add to Cart', 'woocommerce-google-analytics-integration' ) . "'";
-		$parameters['label']    = "($(this).data('product_sku')) ? ($(this).data('product_sku')) : ('#' + $(this).data('product_id'))"; // Product SKU or ID
-
-		if ( ! $this->disable_tracking( $this->ga_enhanced_ecommerce_tracking_enabled ) ) {
-			$item               = '{';
-			$item              .= "'id': ($(this).data('product_sku')) ? ($(this).data('product_sku')) : ('#' + $(this).data('product_id')),";
-			$item              .= "'quantity': $(this).data('quantity')";
-			$item              .= '}';
-			$parameters['item'] = $item;
-
-			$code                   = $this->get_tracking_instance()->tracker_var() . "( 'ec:addProduct', " . $item . ' );';
-			$parameters['enhanced'] = $code;
-		}
-
-		$this->get_tracking_instance()->event_tracking_code( $parameters, '.add_to_cart_button:not(.product_type_variable, .product_type_grouped)' );
-	}
-
-	/**
-	 * Determine if the conditions are met for enhanced ecommerce interactions to be displayed.
-	 * Currently checks if Global Tags OR Universal Analytics are enabled, plus Enhanced eCommerce.
-	 *
-	 * @param  array $extra_checks Any extra option values that should be 'yes' to proceed
-	 * @return bool                Whether enhanced ecommerce transactions can be displayed.
-	 */
-	protected function enhanced_ecommerce_enabled( $extra_checks = [] ) {
-		if ( ! is_array( $extra_checks ) ) {
-			$extra_checks = [ $extra_checks ];
-		}
-
-		// False if gtag and UA are disabled.
-		if ( $this->disable_tracking( $this->ga_use_universal_analytics ) && $this->disable_tracking( $this->ga_gtag_enabled ) ) {
-			return false;
-		}
-
-		// False if gtag or UA is enabled, but enhanced ecommerce is disabled.
-		if ( $this->disable_tracking( $this->ga_enhanced_ecommerce_tracking_enabled ) ) {
-			return false;
-		}
-
-		// False if any specified interaction-level checks are disabled.
-		foreach ( $extra_checks as $option_value ) {
-			if ( $this->disable_tracking( $option_value ) ) {
-				return false;
-			}
-		}
-
-		return true;
-	}
-
-	/**
-	 * Measure a product click and impression from a Product list
-	 *
-	 * @param string     $link The Add To Cart Link
-	 * @param WC_Product $product The Product
-	 */
-	public function track_product( $link, $product ) {
-		if ( $this->enhanced_ecommerce_enabled( $this->ga_enhanced_product_click_enabled ) ) {
-			$this->get_tracking_instance()->listing_impression( $product );
```

Comment:
Yeah that was a bug which meant if tracking product clicks was enabled then we would also track impressions. Again this will probably change quite a bit but for now it's just a simple bugfix.